### PR TITLE
compressor filter: add benchmark

### PR DESCRIPTION
**Pull request by rgs1**
I am debugging a case of slow responses possibly due to compression, so
throwing in some benchmarks to get some perf data about this filter.

```
$ ./bazel-bin/test/extensions/filters/http/common/compressor/compressor_filter_speed_test
2020-03-20 00:15:07
Running ./bazel-bin/test/extensions/filters/http/common/compressor/compressor_filter_speed_test
Run on (8 X 2300 MHz CPU s)
CPU Caches:
  L1 Data 32K (x4)
  L1 Instruction 32K (x4)
  L2 Unified 262K (x4)
  L3 Unified 6291K (x1)
Load Average: 2.17, 2.14, 2.00
***WARNING*** Library was built as DEBUG. Timings may be affected.
---------------------------------------------------------
Benchmark               Time             CPU   Iterations
---------------------------------------------------------
...
compressFull/0/manual_time              14.1 ms         14.3 ms           48
compressFull/1/manual_time              7.06 ms         7.22 ms          104
compressFull/2/manual_time              5.17 ms         5.33 ms          123
compressFull/3/manual_time              15.4 ms         15.5 ms           45
compressFull/4/manual_time              10.1 ms         10.3 ms           69
compressFull/5/manual_time              15.8 ms         16.0 ms           40
compressFull/6/manual_time              15.3 ms         15.5 ms           42
compressFull/7/manual_time              9.91 ms         10.1 ms           71
compressFull/8/manual_time              15.8 ms         16.0 ms           45
compressChunks16384/0/manual_time       13.4 ms         13.5 ms           52
compressChunks16384/1/manual_time       6.33 ms         6.48 ms          111
compressChunks16384/2/manual_time       5.09 ms         5.27 ms          147
compressChunks16384/3/manual_time       15.1 ms         15.3 ms           46
compressChunks16384/4/manual_time       9.61 ms         9.78 ms           71
compressChunks16384/5/manual_time       14.5 ms         14.6 ms           47
compressChunks16384/6/manual_time       14.0 ms         14.1 ms           48
compressChunks16384/7/manual_time       9.20 ms         9.36 ms           76
compressChunks16384/8/manual_time       14.5 ms         14.6 ms           48
compressChunks8192/0/manual_time        14.3 ms         14.5 ms           50
compressChunks8192/1/manual_time        6.80 ms         6.96 ms          100
compressChunks8192/2/manual_time        5.21 ms         5.36 ms          135
compressChunks8192/3/manual_time        14.9 ms         15.0 ms           47
compressChunks8192/4/manual_time        9.71 ms         9.87 ms           68
compressChunks8192/5/manual_time        15.9 ms         16.1 ms           45
...
```

Signed-off-by: Raul Gutierrez Segales <rgs@pinterest.com>

### CODE REVIEW

**Comment by rojkov** [comment]
Thanks! This looks good to me.

BTW, what kind of slowness you're debugging? Bad latency or throughput?

**Comment by rgs1** [comment]
> BTW, what kind of slowness you're debugging? Bad latency or throughput?

Latency -- with big responses (> 100kb)...

**Comment by rgs1** [comment]
Once #10508 lands, I'll update this to exercise different chunk sizes. We can still land this as is, and I'll do a follow-up -- no need to wait.

File: test/extensions/filters/http/common/compressor/compressor_filter_speed_test.cc [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+  for (uint64_t i = 0; i < chunk_count; ++i) {
+    Buffer::OwnedImpl chunk;
+    chunk.add(data);
+    vec.push_back(std::move(chunk));
+  }
+
+  return vec;
+}
+
+static void compressWith(std::vector<Buffer::OwnedImpl> chunks, CompressionParams params,
```
Comment (rojkov) [inline test/extensions/filters/http/common/compressor/compressor_filter_speed_test.cc]: Shouldn't `chunks` be of type `const std::vetcor<Buffer::OwnedImpl>&` to make passing the vector as rvalue meaningful?

Comment (rgs1) [reply test/extensions/filters/http/common/compressor/compressor_filter_speed_test.cc:69]: ah yes (if you meant `const std::vector<Buffer::OwnedImpl>&&`).

**Comment by rgs1** [comment]
PTAL @rojkov @dio

**Comment by rgs1** [comment]
@rojkov @dio one last change -- using the same input for the different chunk sizes, the compression times are mostly the same....

**Comment by rgs1** [comment]
@dio PTAL.

**Comment by rgs1** [comment]
ping?

**Pull request merged**